### ISSUE TITLE
Bug: Not Call My Services Static Function In My Modules.

### ISSUE THREAD

**Issue by yks118** [codeigniter4/CodeIgniter4#5870]
### PHP Version

8.1

### CodeIgniter4 Version

4.1.9

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

fpm-fcgi

### Database

MySQL 8.0.28

### What happened?

Not Call My Service Static Function In My Module.

This issue is caused by "[AppServices](https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/Config/BaseService.php#L192)".

If I modify the source code, it works fine.

/system/Config/BaseService.php In getSharedInstance Static Function.
```diff
...
protected static function getSharedInstance(string $key, ...$params)
{
    ...
-   static::$instances[$key] = AppServices::$key(...$params);
+   static::$instances[$key] = static::$key(...$params);
    ...
}
...
```

After this modification, Services in the "app" folder and "app_my_module" folder work normally.

### Steps to Reproduce

```
app_my_module
├ Config
│　└ Services.php
└ Controllers
```

My Services File.
```php
<?php
namespace AppMyModule\Config;

use CodeIgniter\Config\BaseService;

class Services extends BaseService
{
    public static function myFunction(bool $getShared = true): myClass
    {
        if ($getShared)
            return static::getSharedInstance('myFunction');
        return new myClass();
    }
}
```

My Class File.
```php
...
$myF = \AppMyModule\Config\Service::myFunction();
...
```

### Expected Output

TypeError
AppMyModule\Config\Services::myFunction(): Return value must be of type myClass, null returned

**Comment by kenjis** [comment]
@yks118 Thank you for taking time.

But unfortunately I can't reproduce it.
Is something different from yours?

```diff
--- a/app/Config/Autoload.php
+++ b/app/Config/Autoload.php
@@ -43,6 +43,7 @@ class Autoload extends AutoloadConfig
     public $psr4 = [
         APP_NAMESPACE => APPPATH, // For custom app namespace
         'Config'      => APPPATH . 'Config',
+        'Blog'        => ROOTPATH . 'Blog',
     ];
 
     /**
```

```php
<?php

namespace App\Controllers;

use Blog\MyClass;

class Home extends BaseController
{
    public function index()
    {
        $myClass = new MyClass();
        $myClass->test();
    }
}
```

```php
<?php

namespace Blog\Config;

use Blog\MyClass;
use CodeIgniter\Config\BaseService;

class Services extends BaseService
{
    public static function myFunction(bool $getShared = true): MyClass
    {
        if ($getShared) {
            return static::getSharedInstance('myFunction');
        }

        return new MyClass();
    }
}
```

```php
<?php

namespace Blog;

class MyClass
{
    public function test()
    {
        $myF = \Blog\Config\Services::myFunction();
        dd($myF);
    }
}
```

```
$ php public/index.php 

┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ $myF                                                                                                                            │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
Blog\MyClass (0) ()
═══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
Called from .../Blog/MyClass.php:10 [dd()]
```

**Comment by yks118** [comment]
@kenjis

Hi. Thanks for your reply.

I tried again from the beginning, and I wrote the code according to the answer and confirmed that it works normally.

The problem was that I didn't write it in the Autoload.php file.

This is a problem that occurred while trying to automatically recognize a module instead of writing it in Autoload.php whenever a module is added.

After checking the reply you sent, I solved it in another way.

Thanks for your help.

Sorry for wasting your time.
